src/main/java/com/sap/cloudfoundry/client/facade/adapters/RawCloudServiceBroker.java

```diff
@@ -11,21 +15,26 @@
 public abstract class RawCloudServiceBroker extends RawCloudEntity<CloudServiceBroker> {
 
     @Value.Parameter
-    public abstract Resource<ServiceBrokerEntity> getResource();
+    public abstract ServiceBroker getResource();
 
     @Override
     public CloudServiceBroker derive() {
-        Resource<ServiceBrokerEntity> resource = getResource();
-        ServiceBrokerEntity entity = resource.getEntity();
+        ServiceBroker resource = getResource();
+        String spaceGuid = getSpaceGuid(resource);
         return ImmutableCloudServiceBroker.builder()
                                           .metadata(parseResourceMetadata(resource))
-                                          .name(entity.getName())
-                                          // The password is not returned by the CF controller.
-                                          .password(null)
-                                          .username(entity.getAuthenticationUsername())
-                                          .url(entity.getBrokerUrl())
-                                          .spaceGuid(entity.getSpaceId())
+                                          .name(resource.getName())
+                                          .url(resource.getUrl())
+                                          .spaceGuid(spaceGuid)
                                           .build();
     }
 
+    private String getSpaceGuid(ServiceBroker resource) {
+        return Optional.ofNullable(resource.getRelationships())
+                       .map(ServiceBrokerRelationships::getSpace)
```

Comment:
Is there a possibility that the `relationships` json will be present but the `space` won't?

Reply:
Yes, in case service broker is registered as global, then space is omitted - http://v3-apidocs.cloudfoundry.org/version/3.106.0/index.html#the-service-broker-object